docs/src/main/asciidoc/security-keycloak-admin-client.adoc

```diff
@@ -0,0 +1,261 @@
+to your project by running the following command in your project base directory:
+
+:add-extension-extensions: keycloak-admin-client-reactive,resteasy-reactive-jackson
+include::{includes}/devtools/extension-add.adoc[]
+
+This will add the following to your build file:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-keycloak-admin-client-reactive</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-keycloak-admin-client-reactive")
+implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
+----
+
+We also are going to need a simple resource with a `Keycloak` injected as request scoped CDI bean.
+
+[source,java]
+----
+package org.acme.keycloak.admin.client;
+
+import io.quarkus.security.Authenticated;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.representations.idm.RoleRepresentation;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import java.util.List;
+
+@Authenticated
+@Path("/api/admin")
+public class RolesResource {
+
+        @Inject
+        Keycloak keycloak;
+
+        @GET
+        @Path("/roles")
+        public List<RoleRepresentation> getRoles() {
+            return keycloak.realm("quarkus").roles().list();
+        }
+
+}
+----
+
+Now, you can produce the `Keycloak` bean yourself, or you can instruct Quarkus to produce `Keycloak` bean.
+
+NOTE: If you produce the bean yourself, please remember that `org.keycloak.admin.client.Keycloak` implements `AutoCloseable` and it's important to avoid resource exhaustion and related exceptions. If you configure your beans as explained below, Quarkus will do it for you.
+
+The Keycloak admin client needs access token as its Admin REST API requires authorization.
+If you already have a way to get access token, let's call it `CustomTokenService`, all you need to do is to provide bean that implements `KeycloakAdminClientAuthorizationProvider`, as shown below.
+
+[source,java]
+----
+@ApplicationScoped <1>
+public class CustomAuthZProvider implements KeycloakAdminClientAuthorizationProvider { <2>
+
+    @Inject
+    CustomTokenService customTokenService;
+
+    @Override
+    public String getAuthorization() {
+        return customTokenService.get();
+    }
+
+}
+----
+<1> The provider must be a CDI bean.
+<2> Make sure it implements `KeycloakAdminClientAuthorizationProvider`.
+
+It's also important to tell Quarkus to use the provider and configure the Keycloak server URL.
+
+[source,properties]
+----
+quarkus.keycloak.admin-client."client-name".realm=quarkus <1>
```

Comment:
I'm not sure we need this kind of flexibility to support multiple named KeycloakAdmins, if we ever do then we could do similarly to the way OIDC tenants or OIDC clients are supported but for now I think we should have:
```
quarkus.keycloak.admin-client.client-id=...
```
etc.
Note `client-name` is ok as a property but we already have `client-id` in the OIDC/OIDC Client config...

Reply:
How do you know users in majority of cases want to create exactly one client? There is no extra cost to creating to. `client-name` is the name of the client (named bean).

Reply:
In https://github.com/quarkusio/quarkus/issues/26837#issuecomment-1190837906 you wrote `I believe we'll need to have a KeycloakAdminClient group.` which I though suggest multiple clients too.

Reply:
Please allow me to also provide one else point of view - allowing to configure one client looks good in "quick start"/docs, but in production you will hardly ever have one almighty client with all the powers. You are more likely to give access on need to have basis, which suggest you need more clients for more scenarios. It make sense for users to use our configuration instead of creating the beans themselves as otherwise they need to add some hook to close the client themselves.